### DISCUSSION TITLE
localForage and localsend?

### DISCUSSION THREAD

**Discussion by ghost** [localsend/localsend#86] [Discussion (English, International)] [answered]
Hi everyone.

I wonder if you've all thought about adding support for localsend with [localForage](https://github.com/localForage/localForage).

*what is localsend?*
> "An open source cross-platform alternative to AirDrop"

*what is localForage?*
> "Offline storage, improved. Wraps IndexedDB, WebSQL, or localStorage using a simple but powerful API."

*why?*
1. we can send and receive local data from browsers on different devices with localsend+localForage.
2. open communities like localForage, localsend would mutually benefit
3. LocalSend does not support web yet. so, this would be a good opportunity to have this feature if localsend is needed to send and receive data over a local network with browsing history or non-hosted type applications.

**Comment by Tienisto** [comment, marked as answer]
Thanks for your idea!

Unfortunately, LocalSend does not support web yet.

**Reply by ghost** [reply]
Hi Tienisto.

> Thanks for your idea!
> Unfortunately, LocalSend does not support web yet.

thank you for feedback here. but what do you think of this idea? Is it a good or bad idea?

**Reply by Tienisto** [reply]
I don't have used localForge yet but it seems that it is just a key-value storage. I don't think that you can store large files in them.

**Reply by ghost** [reply]
Hi Tienisto.

> I don't have used localForge yet but it seems that it is just a key-value storage.

thank you for feedback here again.

>  I don't think that you can store large files in them.

If you have time to answer that last question, I'd be happy to hear any feedback here.

- The use case I find interesting is if you have browsing history data across different browsers and devices. If you think about this special use case, you don't need to store very data, it's not very big in terms of storage - so localForage+localsend is interesting.
- Another interesting use case is unhosted applications, which have little data and need synchronization across different devices on a local network. These are two use cases, which I think would be worth the localForage+localsend.

so, my last question is related to these two use cases. what do you think about these two use cases?

**Reply by Tienisto** [reply]
I understand your points. However, to implement this, LocalSend needs to know how to synchronize. Synchronization is not the main focus of LocalSend. LocalSend should only send file from A to B without knowing what B already has. So I think this is out-of-scope of this project.

Synchronizing browser data in the local network is a good idea but it should be implemented using another protocol. Rsync or Syncthing may be good candidates.

**Reply by ghost** [reply]
Hi Tienisto.

> I understand your points. However, to implement this, LocalSend needs to know how to synchronize. Synchronization is not the main focus of LocalSend. LocalSend should only send file from A to B without knowing what B already has. So I think this is out-of-scope of this project.
> Synchronizing browser data in the local network is a good idea but it should be implemented using another protocol. Rsync or Syncthing may be good candidates.

thanks a lot for the feedback. I am happy with your answer, have a great day and week.
